SWENT-GRP5/SOLVIT
# Fix navigation throughout the App

# Purpose:
This pull fixes the incorrect behavior of navigation throughout the app

- **What issue does it solve?** This PR solves the issue when navigating throughout the app and using the go-back button.

### Related Issues:
- #247 

# Changes Made:
- Make sure that the current screen is always stored as the previous when navigating through the app withe the use of `navigationActions.navigateTo()`
- Modified `Calendar.kt` and `ProviderProfile.kt` backButton behavior to goBack()

# How to Test:
1. Run the app on a phone or emulator
2. Navigate through all screens.
3. Click on the goBack button
4. Check that go back button works well every time

# Comments:
- `EditSeekerProfile.kt` is addressed in another PR

---

# Checklist:
